tests/functional/metrics/test_metrics_provider.py

```diff
@@ -1,101 +1,113 @@
+        )
 
-@pytest.fixture
-def metrics_class() -> BaseProvider:
-    class MetricsClass(BaseProvider):
-        def __init__(self, provider):
-            self.provider = provider
-            super().__init__()
 
-        def add_metric(self, name: str, value: float, tag: List = None, *args, **kwargs):
-            self.provider.add_metric(name=name, value=value, tag=tag)
+class FakeMetricsClass:
+    def __init__(self, provider: FakeMetricsProvider | None = None):
+        if provider is None:
+            self.provider = FakeMetricsProvider()
+        else:
+            self.provider = provider
 
-        def flush_metrics(self, raise_on_empty_metrics: bool = False) -> None:
-            self.provider.serialize_metric_set(raise_on_empty_metrics=raise_on_empty_metrics)
-            self.provider.flush()
-            self.provider.clear()
+    def add_metric(self, name: str, value: float, tag: List = None, *args, **kwargs):
+        self.provider.add_metric(name=name, value=value, tag=tag)
 
-        def add_cold_start_metric(self, context: Any) -> None:
-            self.provider.add_metric(name="ColdStart", value=1, function_name=context.function_name)
+    def flush_metrics(self, raise_on_empty_metrics: bool = False) -> None:
+        self.provider.flush_metrics(raise_on_empty_metrics=raise_on_empty_metrics)
 
-        def serialize_metric_set(self, raise_on_empty_metrics: bool = False, *args, **kwargs):
-            if raise_on_empty_metrics and len(self.metric_store) == 0:
-                raise SchemaValidationError("Must contain at least one metric.")
+    def add_cold_start_metric(self, context: LambdaContext) -> None:
+        self.provider.add_cold_start_metric(context=context)
 
-            self.result = self.provider.flush()
+    def serialize_metric_set(self, raise_on_empty_metrics: bool = False, *args, **kwargs):
+        self.provider.serialize_metric_set(raise_on_empty_metrics=raise_on_empty_metrics)
 
-        def clear_metrics(self) -> None:
-            self.provider.clear_metrics()
+    def clear_metrics(self) -> None:
+        self.provider.clear_metrics()
 
-    return MetricsClass
+    def log_metrics(
+        self,
+        lambda_handler: Callable[[Dict, Any], Any] | Optional[Callable[[Dict, Any, Optional[Dict]], Any]] = None,
+        capture_cold_start_metric: bool = False,
+        raise_on_empty_metrics: bool = False,
+        default_dimensions: Dict[str, str] | None = None,
+    ):
+        return self.provider.log_metrics(
+            lambda_handler=lambda_handler,
+            capture_cold_start_metric=capture_cold_start_metric,
+            raise_on_empty_metrics=raise_on_empty_metrics,
+        )
 
 
-def test_metrics_provider_basic(capsys, metrics_provider, metric):
-    provider = metrics_provider()
-    provider.add_metric(**metric)
-    provider.serialize_metric_set()
-    provider.flush()
+def test_metrics_class_with_default_provider(capsys, metric):
```

Comment:
we don't need it anymore as we have this tested under functional metrics `Metrics()` ;)